Assets/00_Content/Scripts/Vikings/Viking.cs

```diff
@@ -112,5 +169,76 @@ public class Viking : Interactable {
 		public override bool CanInteract(GameObject player, PickUp item) {
 			return state.CanInteract(player, item);
 		}
+
+		private void OnTriggerEnter(Collider other) {
+			if (weaponLayer.ContainsLayer(other.gameObject.layer)) {
+				RegisterHitFromPlayer(other);
+			}
+		}
+
+		private void RegisterHitFromPlayer(Collider other) {
+
+			if (state is LeavingVikingState)
+				return;
+
+			//TODO - Probably get some more generic type of weapon instead of axe. Make an interface for all weapons?
+			Axe axe = other.gameObject.GetComponent<Axe>();
+
+			if (axe.IsAttacking && !isAttacked) {
+
+				isAttacked = true;
+
+				foreach (MeshRenderer hitHighlightMeshRenderer in hitHighlightMeshRenderers) {
+					hitHighlightMeshRenderer.material = brawlingMaterial;
+				}
+
+				if (!IsSeated) {
+					PlayerComponent playerComponent = axe.GetComponentInParent<PlayerComponent>();
+					Vector3 direction = (playerComponent.transform.position - transform.position).normalized;
+					navMeshAgent.enabled = false;
+					rb.isKinematic = false;
+					rb.AddForce(direction * axe.WeaponData.knockBackStrength * -1, ForceMode.Impulse);
+				}
+
+				StartCoroutine(ResetHitSimulation());
+
+				if (state is BrawlingVikingState) {
+					Stats.TakeBrawlDamage(axe.WeaponData.brawlDamage);
+					if (Stats.BrawlHealth <= 0) {
+
+						ChangeState(new LeavingVikingState(this));
+						return;
+					}
+				}
+
+				if (state is WaitingForSeatVikingState) {
+					ChangeState(new LeavingVikingState(this));
+					return;
+				}
+
+				Stats.TakeMoodDamage(axe.WeaponData.moodDamage);
+
+				if (Stats.Mood <= 15f) {
+					ChangeState(new BrawlingVikingState(this, axe.GetComponentInParent<PlayerComponent>()));
+				}
+			}
+		}
```

Comment:
Would having a hit method or something on the different VikingStates fit here? Does make it easier to control how each state reacts to being hit. Could also have the weapon/axe call the hit method with a force instead of calculating it here.

Reply:
Yeah that´s a good solution! thanks!

Reply:
Did as you said, but kept the force calculation/logic here.